Tests/UI/BillingForm/View/ExpiryDateFormatterTests.swift

```diff
@@ -29,184 +29,166 @@ final class ExpiryDateFormatterTests: XCTestCase {
         let testInput = "14"
         let formatter = ExpiryDateFormatter()
-        let output = formatter.formatForDisplay(input: testInput)
+        let displayString = formatter.formatForDisplay(input: testInput)
 
-        XCTAssertNil(output.displayString)
-        XCTAssertNil(output.validationError)
+        XCTAssertNil(displayString)
     }
 
     func testValidMonthInput() {
         let testInput = "11"
         let formatter = ExpiryDateFormatter()
-        let output = formatter.formatForDisplay(input: testInput)
+        let displayString = formatter.formatForDisplay(input: testInput)
 
-        XCTAssertEqual(output.displayString, "11/")
-        XCTAssertNil(output.validationError)
+        XCTAssertEqual(displayString, "11/")
     }
 
     func testValidMonthInputWithSlash() {
         let testInput = "11/"
         let formatter = ExpiryDateFormatter()
-        let output = formatter.formatForDisplay(input: testInput)
+        let displayString = formatter.formatForDisplay(input: testInput)
 
-        XCTAssertEqual(output.displayString, "11/")
-        XCTAssertNil(output.validationError)
+        XCTAssertEqual(displayString, "11/")
     }
 
     func testValidMonthAndPastDecadeInput() {
         let testInput = "11/1"
         let formatter = ExpiryDateFormatter()
-        let output = formatter.formatForDisplay(input: testInput)
+        let displayString = formatter.formatForDisplay(input: testInput)
 
-        XCTAssertEqual(output.displayString, "11/1")
-        XCTAssertEqual(output.validationError, .inThePast)
+        XCTAssertEqual(displayString, "11/1")
     }
 
     func testValidMonthAndFutureDecadeDigitInput() {
         let testInput = "11/8"
         let formatter = ExpiryDateFormatter()
-        let output = formatter.formatForDisplay(input: testInput)
+        let displayString = formatter.formatForDisplay(input: testInput)
 
-        XCTAssertEqual(output.displayString, "11/8")
-        XCTAssertNil(output.validationError)
+        XCTAssertEqual(displayString, "11/8")
     }
 
     func testValidMonthAndFullPastYearInput() {
         let testInput = "11/21"
         let formatter = ExpiryDateFormatter()
-        let output = formatter.formatForDisplay(input: testInput)
+        let displayString = formatter.formatForDisplay(input: testInput)
 
-        XCTAssertEqual(output.displayString, "11/21")
-        XCTAssertEqual(output.validationError, .inThePast)
+        XCTAssertEqual(displayString, "11/21")
     }
 
     func testValidMonthAndFullValidYearInput() {
         let testInput = "11/88"
         let formatter = ExpiryDateFormatter()
-        let output = formatter.formatForDisplay(input: testInput)
+        let displayString = formatter.formatForDisplay(input: testInput)
 
-        XCTAssertEqual(output.displayString, "11/88")
-        XCTAssertNil(output.validationError)
+        XCTAssertEqual(displayString, "11/88")
     }
 
     func testIncompleteMonthAndFullYearInput() {
```

Comment:
Is there an updated test case to test the Current Year + Previous Months of the Current year? If so can you please point me to that.

Reply:
There is no necessity for such a test case, since this is the formatter and no longer does the date validation. That test will reside in the next PR (`CardValidatorTests`) and it's been working from the beginning. So, there is no change in this PR to make the expiry date validation work, but, only to make the code flow through the required code pieces. 

Which means, `ExpiryDateFormatter` was doing the validation check, that it shouldn't have been doing from the beginning. And the expiry date validation logic was broken because of that. Now, after it stopped validating it but only kept formatting it, there must be no case that makes difference in its test cases. I mean, `11/88` and `05/23` makes no difference since those two inputs are in the same format. Adding a test case for it will be redundant. @deepesh-vasthimal-cko